File: uiapplication/src/main/java/com/futing/uiapplication/clickevent/MainActivity.java
package com.futing.uiapplication.clickevent;


import android.os.Bundle;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import com.futing.uiapplication.R;

public class MainActivity extends AppCompatActivity {

    private Button btn_click;

    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_click);

        btn_click = findViewById(R.id.btn_click);

        btn_click.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.e(TAG, "onClick");
            }
        });
        //return true以后   Log.e(TAG, "onClick");不会打印   true以后
        btn_click.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View v, MotionEvent event) {
                Log.e(TAG, "onTouch: " + event.getAction());
//                return true; // 不会打印   Log.e(TAG, "onClick");
                return false; // 会打印   Log.e(TAG, "onClick");
            }
        });
    }
}

File: kotlinlib/src/main/java/com/example/kotlinlib/MainActivity.kt
package com.example.kotlinlib

import android.os.Bundle
//import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.kotlinlib.coverkotlin.User
import com.example.kotlinlib.coverkotlin.UserNullValue1
import com.example.kotlinlib.coverkotlin.UserNullable
import com.example.kotlinlib.objectcase.NetworkManager
import com.example.kotlinlib.objectcase.Repository
import kotlinx.coroutines.MainScope
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        Android Jetpack Compose中用于实现沉浸式状态栏的一种方法
//        enableEdgeToEdge()
//        直接new一个kotlin类型的eactivity的话 UI是layoutd



        //var lastName: String? = null 要带问好才行要不然无法赋值
//        val user = User("futing")
        val user = User("futing","jingjing")

        val userNULL = UserNullable("jane") //same as UserNullableJane", null)
        val userNULL1 = UserNullable("jane",null) //same as User("Jane", null)
        val userNULLVaule = UserNullValue1(lastName = "futing") //same as User(null, "Doe")
        val userNULLVaule1 = UserNullValue1(null, "futing")

        val users = Repository.instance?.getUser()
//        val userNames = Repository.instance?.formattedUserNames()

        //伴生对象使用
        NetworkManager.performRequest("https://blog.csdn.net/qq_42751010/article/details/134826937")

        setContentView(R.layout.activity_main_kotlin)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }


    }

    fun coroutines(){
        //默认在UI线程中启动协程
        MainScope().launch {


        }

    }
}